test/auth_aws/test_auth_aws.py

```diff
@@ -111,6 +111,67 @@ def test_poisoned_cache(self):
         client.get_database().test.find_one()
         self.assertNotEqual(auth.get_cached_credentials(), None)
 
+    def test_environment_variables_ignored(self):
```

Comment:
Is this behavior outlined in the spec? Shouldn't the env vars take precedence even if we already have a previous cached credential?

Reply:
Discussed in person, this behavior happens because we rely on aws for the credential lookup. The code is like:
```
if cached_creds:
    return cached_creds
creds = get_aws_creds()
if should_cache(creds):
    cached_creds = creds
return creds
```

So if the app is always using env vars then there will never be a cached credential and dynamic env var changes will propagate immediately. However, if the app is using temp creds then later tries to switch to dynamic env var creds, then changes won't propagate until the cached creds are cleared.